cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt

```diff
@@ -540,31 +542,50 @@ private class CampaignManagerImpl(
         }
 
     override val hasLocalData: Boolean
-        get() = dataStorage.gdprConsentStatus != null || dataStorage.usPrivacyString != null
-
-    override fun handleMetaDataLogic(md: MetaDataResp?) {
-        metaDataResp = md
-        md?.let {
-            it.ccpa?.apply {
-                applies?.let { i -> dataStorage.ccpaApplies = i }
-                sampleRate?.let { i ->
-                    if (i != dataStorage.ccpaSamplingValue) {
-                        dataStorage.ccpaSamplingValue = i
-                        dataStorage.ccpaSamplingResult = null
-                    }
+        get() = dataStorage.gdprConsentStatus != null || dataStorage.uspstring.isNullOrEmpty().not()
+
+    override fun handleMetaDataResponse(response: MetaDataResp?) {
+        // update meta data response in the data storage
+        metaDataResp = response
+
+        if (response == null) return
+
+        // handle ccpa
+        response.ccpa?.apply {
+            applies?.let { i ->
```

Comment:
@bohdan-go-wombat could you rename `i` to something more intuitive in the lines below? Otherwise I don't see the improvement using `i` over `it`.

Reply:
@andresilveirah agree